test/functional/p2p_add_connections.py

```diff
@@ -0,0 +1,94 @@
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+
+        self.log.info("Add 2 block-relay-only connections to node 1")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[1].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+
+        self.log.info("Add some inbound connections to node 1")
+        for i in range(5):
+            self.log.info("inbound: {}".format(i))
+            self.nodes[1].add_p2p_connection(P2PInterface())
+
+        self.log.info("Add 8 outbounds to node 1")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[1].add_outbound_p2p_connection(P2PInterface())
+
+        self.log.info("Check the connections opened as expected")
+        assert_equal(len(self.nodes[0].getpeerinfo()), 10)
+        assert_equal(len(self.nodes[1].getpeerinfo()), 15)
+
+        # check node0 has all outbound connections
+        node0_peers = []
+        for i in range(10):
+            node0_peers.append(self.nodes[0].getpeerinfo()[i]['inbound'])
+        assert_equal(node0_peers.count(False), 10)
+
+        # check node1 has 10 outbounds & 5 inbounds
+        node1_peers = []
+        for i in range(15):
+            node1_peers.append(self.nodes[1].getpeerinfo()[i]['inbound'])
+
+        assert_equal(node1_peers.count(False), 10)
+        assert_equal(node1_peers.count(True), 5)
+
+        self.log.info("Disconnect p2p connections & try to re-open")
+        self.nodes[0].disconnect_p2ps()
+
+        self.log.info("Add 8 outbounds to node 0")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+
+        self.log.info("Restart node 0 and try to reconnect to p2ps")
+        self.restart_node(0)
+
+        self.log.info("Add 8 outbounds to node 0")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
```

Comment:
50720f411e05cf1bba added these as sanity checks. Could add a helper test that checks the passed number of in/out connections for each node, e.g. `self.test_conns([(0, 10), (5, 10)])` to assert these convienently 
```diff
         self.log.info("Disconnect p2p connections & try to re-open")
         self.nodes[0].disconnect_p2ps()
+        assert_equal(self.nodes[0].getnetworkinfo()["connections_out"], 0)
+        assert_equal(self.nodes[1].getnetworkinfo()["connections_in"], 5)
+        assert_equal(self.nodes[1].getnetworkinfo()["connections_out"], 10)
 
         self.log.info("Add 8 outbounds to node 0")
         for i in range(8):
             self.log.info("outbound: {}".format(i))
             self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+        assert_equal(self.nodes[0].getnetworkinfo()["connections_out"], 8)
+        assert_equal(self.nodes[1].getnetworkinfo()["connections"], 15)
 
         self.log.info("Add 2 block-relay-only connections to node 0")
         for i in range(2):
             self.log.info("block-relay-only: {}".format(i))
             self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+        assert_equal(self.nodes[0].getnetworkinfo()["connections_out"], 10)
+        assert_equal(self.nodes[1].getnetworkinfo()["connections"], 15)
 
         self.log.info("Restart node 0 and try to reconnect to p2ps")
         self.restart_node(0)
@@ -83,11 +82,15 @@ class P2PAddConnections(BitcoinTestFramework):
         for i in range(8):
             self.log.info("outbound: {}".format(i))
             self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+        assert_equal(self.nodes[0].getnetworkinfo()["connections_out"], 8)
+        assert_equal(self.nodes[1].getnetworkinfo()["connections"], 15)
 
         self.log.info("Add 2 block-relay-only connections to node 0")
         for i in range(2):
             self.log.info("block-relay-only: {}".format(i))
             self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+        assert_equal(self.nodes[0].getnetworkinfo()["connections_out"], 10)
+        assert_equal(self.nodes[1].getnetworkinfo()["connections"], 15)
 ```

Reply:
👍 added a helper to compare connections using `getnetworkinfo` and added more checks.